src/client/Client.cc

```diff
@@ -8762,19 +8756,12 @@ int Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl)
   if (f->flags & O_DIRECT)
     have &= ~CEPH_CAP_FILE_CACHE;
 
-  Mutex uninline_flock("Client::_read_uninline_data flock");
-  Cond uninline_cond;
-  bool uninline_done = false;
-  int uninline_ret = 0;
-  Context *onuninline = NULL;
+  std::unique_ptr<C_SaferCond> onuninline = nullptr;
```

Comment:
You can actually leave it unassigned. The default constructor initializes `std::unique_ptr` with a `nullptr`.

Reply:
Great! actually, i was entangled with this two place you pointed out, thanks.